src/components/EditEventForm/index.tsx

```diff
@@ -24,17 +24,34 @@ const months = [
   'December',
 ];
 
-const days = [];
+const days: number[] = [];
 for (let i = 1; i <= 31; i += 1) {
   days.push(i);
 }
 
-const hours = [];
+const hours: number[] = [];
 for (let i = 1; i <= 12; i += 1) {
   hours.push(i);
 }
 
-const EditEventForm = (props) => {
+interface EventProp {
+  start: string,
+  end: string,
+  [key: string]: any
+}
+
+interface EditEventFormProps {
+  event: EventProp
+  deleteEvent: Function
+  setFieldValue: Function
+  setFieldTouched: Function
+  handleBlur: FocusEventHandler
+  handleChange: ChangeEventHandler
+  handleSubmit: FormEventHandler
+  values: {[key: string]: any}
```

Comment:
Only commenting with the clarification that specific types for action payloads should (and most likely will) be added once the backend team constructs a type library to couple with the API.

Reply:
how would we sync the types with backend though

Reply:
We could sync with backend in a couple of ways. Here's a list of all the options with no personal preference in particular:
- Monorepo the whole membership portal and keep the API model in a separate project folder. You can use TypeScript [project references](https://www.typescriptlang.org/docs/handbook/project-references.html) to share types between the projects.
- Create a separate "types" package that can be linked to both the API and the UI with `npm link`. Then, we update types in the third repository and import them into each project's `tsconfig.json`.
- Separate "types" repository again, although this time use Git submodules instead of `npm`. Slightly less fun to implement, though.
- Installing both projects when developing for either and using some symlinking when cloning the repos to get the types synced between one another.

Monorepo probably works best for absolute complete synchronization of types, especially since our entire stack will eventually be in TS, but that's debatably harder than the separate types package.